packages/mastering/src/mastering/index.ts

```diff
@@ -550,7 +553,12 @@ function shuffleAnswerOptions(exam: Exam, multichoiceShuffleSecret: string) {
 async function renderFormulas(exam: Element, throwOnLatexError?: boolean) {
   for (const formula of exam.find<Element>('//e:formula', ns)) {
     try {
-      const { svg, mml } = await renderFormula(formula.text(), getAttribute('mode', formula, null), throwOnLatexError)
+      // Load render-formula lazily, since initializing mathjax-node is very expensive.
+      const { svg, mml } = await require('./render-formula')(
```

Comment:
Pitäisikö tämä laiska lataus tehdä for-loopin ulkopuolella tai laittaa johonkin loopin kierrosten välillä jaettuun muuttujaan, vai onko require niin nopea että ok tehdä jokaiselle kaavalle erikseen?

Reply:
En usko, että sillä on merkitystä. Node kakuttaa requiren palauttamat moduulit, ja sinänsä tossa renderöinnissä menee tyyliin ~10ms per kaava (MathJax on tosi hidas 😿 ).

Yksinkertainen mikrobenchmark kertoo seuraavaa:

```
 $ cat benchmark.js
const Benchmark = require('benchmark')

new Benchmark.Suite()
  .add('require()', () => { require('util') })
  .on('cycle', e => { console.log(String(e.target)) })
  .run({ async: true })

 $ node benchmark.js
require() x 8,290,676 ops/sec ±0.80% (89 runs sampled)
```

eli noin 120 nanosekuntia täysin optimoituna

Reply:
Kiitos, Mathjax...